connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroUtils.scala

```diff
@@ -197,19 +197,21 @@ private[sql] object AvroUtils extends Logging {
 
     def hasNextRow: Boolean = {
       while (!completed && currentRow.isEmpty) {
-        if (fileReader.pastSync(stopPosition)) {
+        // In the case of empty blocks in an Avro file, `blockRemaining` could still read as 0 so
+        // `fileReader.hasNext()` returns false but advances the cursor to the next block, so we
+        // need to call `fileReader.hasNext()` again to correctly report if the next record
+        // exists.
```

Comment:
hmmm, what if there are continuous empty blocks? how many times we need to call `hasNext`?

Reply:
That's what I was thinking but I could not find an example that double hasNext has not fixed. So it seems to be fine. I just could not find a better way of solving this issue at the moment given the FileReader interface.